docs/src/guides/scalar_index.md

```diff
@@ -0,0 +1,90 @@
+# Building Scalar Index
+
+Similar to many SQL databases, LanceDB supports several types of Scalar indices to accelerate search
+over scalar columns.
+
+- `BTREE`: The most common type is BTREE. This index is inspired by the btree data structure
+  although only the first few layers of the btree are cached in memory.
+  It will perform well on columns with a large number of unique values and few rows per value.
+- `BITMAP`: this index stores a bitmap for each unique value in the column.
+  This index is useful for columns with a small number of unique values and many rows per value.
+  For example, a column presents `categories`. In Apache Arrow terminalogy, a column can be presented
+  by `DictionaryType`.
+- `LABEL_LIST`: a special index that is used to index list columns whose values have small cardinality.
+  For example, a column that contains lists of tags (e.g. `["tag1", "tag2", "tag3"]`) can be indexed with a `LABEL_LIST` index.
+  This index can only speedup queries with `array_has_any()` or `array_has_all()` filters.
+
+=== "Python"
+
+    ```python
+    import lancedb
+    db = lancedb.connect("./db")
+    table = db.open_table("my_table")
+    table.create_scalar_index("category", index_type="BITMAP")
+    ```
+
+=== "Typescript"
+
+    Only `BTree` index is supported today. `BITMAP` and `LABEL_LIST` will be added soon.
```

Comment:
Do we have a GH issue with this? We should create one if not. Might be nice to link to that issue here so users can track the progress of that or maybe see it needs doing and pick it up.

Reply:
Added.

Reply:
Adding python lancedb api here:

https://github.com/lancedb/lancedb/pull/1495/files#diff-1fd270f2427efc004849c52acb4f60a2d49396cec44dacea2a553455d820acbaR100